tron/core/actionrun.py

```diff
@@ -184,7 +192,7 @@ def check_state(self, state):
     @classmethod
     def from_state(
         cls, state_data, parent_context, output_path,
-        job_run_node, cleanup=False,
+        job_run_node, cleanup=False, retries_remaining=None, exit_statuses=None,
```

Comment:
Why not use retries_remaining and exit_statuses from the state_data, so you don't have to pass it separately?

Otherwise, you'll need to edit ActionRunFactory.action_run_from_state.